src/scripts/ui/MarketBuildingBody.tsx

```diff
@@ -48,7 +48,11 @@ export function MarketBuildingBody({ gameState, xy }: IBuildingComponentProps):
       const buyAmount = getMarketBuyAmount(sellResource, sellAmount, buyResource, xy, gameState);
       const sellValue = Config.ResourcePrice[sellResource]! * sellAmount;
       const buyValue = Config.ResourcePrice[buyResource]! * buyAmount;
-      tradeValues.set(sellResource, buyValue / sellValue - 1);
+      let tradeValue = buyValue / sellValue - 1;
+      if (!Number.isFinite(tradeValue)) {
```

Comment:
This is hiding the problem instead of solving it. So instead of checking whether `tradeValue` is a number of not, it's worth figuring our why `tradeValue` is not a finite number in the first place. From what I can see, `sellValue = 0` would cause this? And why `sellValue` is 0 in the first place? (It should never be zero)

Reply:
This is caused when you shut down a market.  and `capacity = 0` so `sellAmout` from `getMarketSellAmount` is returned as 0.

Reply:
Make sense. So probably `getMarketSellAmount` should simply ignore capacity.